test/lib/net/dhcp_test.rb

```diff
@@ -92,16 +92,30 @@ class DhcpTest < ActiveSupport::TestCase
     refute record1.valid?
   end
 
-  test "dhcp record validation should fail on MAC conflict" do
+  test "dhcp record must not validate when there is IP conflict" do
+    ProxyAPI::Resource.any_instance.stubs(:get).with("127.0.0.0/aa:bb:cc:dd:ee:01").returns(@lease1)
     ProxyAPI::Resource.any_instance.stubs(:get).with("127.0.0.0/127.0.0.1").returns(@lease1)
+    ProxyAPI::Resource.any_instance.stubs(:get).with("127.0.0.0/aa:bb:cc:dd:ee:02").raises(RestClient::ResourceNotFound, 'Record not found')
     record1 = Net::DHCP::Record.new(:hostname => "discovered_host1", :mac => "aa:bb:cc:dd:ee:02",
                                     :network => "127.0.0.0", :ip => "127.0.0.1",
                                     "proxy" => subnets(:one).dhcp_proxy)
     refute record1.conflicts.empty?
+    refute record1.valid?
+  end
+
+  test "dhcp record must not validate when there is MAC conflict" do
+    ProxyAPI::Resource.any_instance.stubs(:get).with("127.0.0.0/aa:bb:cc:dd:ee:01").returns(@lease1)
+    ProxyAPI::Resource.any_instance.stubs(:get).with("127.0.0.0/127.0.0.1").returns(@lease1)
+    ProxyAPI::Resource.any_instance.stubs(:get).with("127.0.0.0/127.0.0.2").raises(RestClient::ResourceNotFound, 'Record not found')
+    record1 = Net::DHCP::Record.new(:hostname => "discovered_host1", :mac => "aa:bb:cc:dd:ee:01",
+                                    :network => "127.0.0.0", :ip => "127.0.0.2",
+                                    "proxy" => subnets(:one).dhcp_proxy)
+    refute record1.conflicts.empty?
+    refute record1.valid?
   end
 
-  test "dhcp record validation on multiple discovered leases" do
-    lease2 = '{
+  test "dhcp record must validate with multiple leases with same MAC" do
```

Comment:
adding an `assert record1.valid?` to match the other tests would be useful here too